message_processor.go

```diff
@@ -45,6 +48,12 @@ This is a routine to decode the log stream into LogEntry and store it in buffer
 and send the buffer to splunk logger and json logger
 */
 func (mg messageProcessor) consumeLog(lf *logPair) {
+	// Check env variable for json logs
+	jsonLogs, err := strconv.ParseBool(os.Getenv(envVarJSONLogs))
```

Comment:
are we pulling this value from an OS environment variable or are we setting up a switch from a param? I'd follow the same way we do buffer variable set/get.

Reply:
OS environment, same as buffer settings

Reply:
This seems off we should be using parameters read in to plugin. The buffer does not use os.Getenv to retrieve this value. just reference variable direct

`jsonLogs, err := strconv.ParseBool(envVarJSONLogs)`

Reply:
Created getAdvancedOptionBool function in splunk_logger.go to handle jsonLogs